modules/stochastic_tools/include/distributions/NormalDistribution.h

```diff
@@ -10,20 +10,49 @@
 #ifndef NORMALDISTRIBUTION_H
 #define NORMALDISTRIBUTION_H
 
-#include "BoostDistribution.h"
+#include "Distribution.h"
 
 class NormalDistribution;
 
 template <>
 InputParameters validParams<NormalDistribution>();
 
 /**
- * A class used to generate Normal distribution via Boost
+ * A class used to generate a normal distribution
+ * This differs from the MOOSE version in that it provides a method to set the parameters
+ * after it has been constructed.
  */
-class NormalDistribution : public BoostDistribution<boost::math::normal_distribution<Real>>
+class NormalDistribution : public Distribution
 {
 public:
   NormalDistribution(const InputParameters & parameters);
+
+  virtual Real pdf(const Real & x) override;
+  virtual Real cdf(const Real & x) override;
+
+  /**
+   * Compute the quantile
+   * @param p The cdf for which the quantile is evaluated
+   **/
+  virtual Real quantile(const Real & p) override;
+
+  /**
+   * Set parameters for this distribution
+   * @param mean The mean of the distribution
+   * @param standard_deviation The standard deviation of the distribution
+   **/
+  void setParams(const Real mean, const Real standard_deviation);
```

Comment:
@aeslaughter This is probably the most controversial part of this commit. I added these `setParams()` methods because I was re-using the same distribution objects lots of times with different parameters in my sampler class. Maybe there's a preferred way of doing this, but I really need this kind of capability. I have my own one-off versions of these objects, and they're working just fine, but if you're OK with this (or some other way of being able to modify those parameters on the fly), then I can get rid of them and use these instead.

Reply:
I don't like the idea of changing the _mean and _standard_deviation. That opens up the possibility of one object that uses a distribution to change the state of the object that another could be using.

What I would prefer, will take a little more work, is that the pdf/cdf/quantile methods have an overloaded versions: one that requires the mean, std_dev to be passed in, with the other using the values supplied as input parameters. The second version can call the first.

Also, the get methods in the DistributionInterface should return const references. I am surprised they were not originally, so there might be a reason why there are not. If for some reason they cannot be made constant directly I would be onboard for making these methods mutable to that the user can't do something like you did and alter state/behavior for other objects using the distribution.

Is this something you need me to work on, I wouldn't mind? I can add this to the list with the transfer on final. I can do it next week, first thing, I promise.

Reply:
Different distributions have different number of parameters. Does that mean we need to implement the overloaded versions for each distribution type?

Reply:
Yes, but I would rather do that and have the interface correct then allow for unstable behavior. Creating distributions is not something that will be done all that often, so it is worth the work.

We could likely use use C+11 variadic templates to make a single method, but that might be more trouble than it is worth and make understanding how to use the functions more difficult.

Reply:
If you do not mind, I could remove those ``setParams`` from this PR and leave the rest work to you. :-) Thanks!

Reply:
@aeslaughter Having overloaded versions of the pdf/cdf/quantile methods where you pass the parameters in makes sense to me. I'm only actually using the quantile method in this way, so that's the only one I care about.

With what you are proposing, it will all work fine if the distribution objects are const. I'm actually not even using the DistributionInterface to get the distribution objects, because it doesn't make any sense for the user to define these distributions in the input file. I'm creating the ones I need in the code (using FEProblem.addDistribution()), and then getting pointers to them using FEProblem.getDistribution().

I would propose that we just strip out all of these `setParams()` methods from this commit, and then add these variants of the methods where we pass the parameters in later on.  As part of that, we should make all of the member variables that these methods are manipulating const.

Reply:
In case you're interested in how I'm using these, you can look at RPVFractureSampler.C in Grizzly.